provider/cluster_rosa_classic_resource.go

```diff
@@ -623,6 +643,154 @@ func createClassicClusterObject(ctx context.Context,
 	return object, err
 }
 
+func (r *ClusterRosaClassicResource) validateAccountRoles(ctx context.Context, state *ClusterRosaClassicState) error {
+	r.logger.Debug(ctx, "Validating if cluster version is compatible to account roles' version")
+	region := state.CloudRegion.Value
+	version := ""
+	if !state.Version.Unknown && !state.Version.Null {
+		version = state.Version.Value
+	}
+
+	if version == "" {
+		versionList, err := r.getVersionList(r.logger, ctx)
+		if err != nil {
+			return err
+		}
+		version = versionList[0]
+	}
+
+	r.logger.Debug(ctx, "Cluster version is %s", version)
+	roleARNs := []string{
+		state.Sts.RoleARN.Value,
+		state.Sts.SupportRoleArn.Value,
+		state.Sts.InstanceIAMRoles.MasterRoleARN.Value,
+		state.Sts.InstanceIAMRoles.WorkerRoleARN.Value,
+	}
+
+	for _, ARN := range roleARNs {
+		if ARN == "" {
+			continue
+		}
+		// get role from arn
+		role, err := getRoleByARN(ARN, region)
+		if err != nil {
+			return fmt.Errorf("Could not get Role '%s' : %v", ARN, err)
+		}
+
+		validVersion, err := hasCompatibleVersionTags(r.logger, ctx, role.Tags, getOcmVersionMinor(version))
+		if err != nil {
+			return fmt.Errorf("Could not validate Role '%s' : %v", ARN, err)
+		}
+		if !validVersion {
+			return fmt.Errorf("account role '%s' is not compatible with version %s. "+
+				"Run 'rosa create account-roles' to create compatible roles and try again",
+				ARN, version)
+		}
+	}
+
+	return nil
+}
+func hasCompatibleVersionTags(logger logging.Logger, ctx context.Context, iamTags []*iam.Tag, version string) (bool, error) {
+	if len(iamTags) == 0 {
+		return false, nil
+	}
+	for _, tag := range iamTags {
+		if aws.StringValue(tag.Key) == tagsOpenShiftVersion {
+			logger.Debug(ctx, "role version is %s", aws.StringValue(tag.Value))
+			if version == aws.StringValue(tag.Value) {
+				return true, nil
```

Comment:
Why do we return here `true`?

Reply:
The same as you're doing in rosa cli: https://github.com/openshift/rosa/blob/master/pkg/aws/policies.go#L413
It means that both of them are holding the same tag version